smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/middleware/MiddlewareRenderable.kt

```diff
@@ -22,5 +27,5 @@ interface MiddlewareRenderable {
 
     val position: MiddlewarePosition
 
-    fun render(model: Model, symbolProvider: SymbolProvider, writer: SwiftWriter, op: OperationShape, operationStackName: String)
+    fun render(model: Model, symbolProvider: SymbolProvider, writer: SwiftWriter, op: OperationShape, operationStackName: String, executionContext: MiddlewareRenderableExecutionContext)
```

Comment:
does it make sense to default this to CLIENT since in most situations its CLIENT? then you can avoid passing it everywhere?